js/client-battle-tooltips.js

```diff
@@ -1152,6 +1165,45 @@ var BattleTooltips = (function () {
 			basePower = Math.floor(basePower * abilityBoost);
 			basePowerComment = this.makePercentageChangeText(abilityBoost, ability);
 		}
+
+		var allyActive = pokemon.side.active;
+		var auraBoosted = '';
+		var auraBroken = false;
+		if (move.category !== 'Status' && allyActive.length > 1) {
+			for (var i = 0; i < allyActive.length; i++) {
+				var ally = allyActive[i];
+				if (!ally || ally.fainted) continue;
+				if (!auraBoosted && ((ally.ability === 'Fairy Aura' && move.type === 'Fairy') || (ally.ability === 'Dark Aura' && move.type === 'Dark'))) {
```

Comment:
This is way too complicated of a condition and would probably benefit from being simplified in some way.

My first thought is that `!auraBoosted &&` isn't really a necessary condition. It's a micro-optimization that makes the code less readable so I'd probably just leave it off entirely.

Reply:
My second thought is just to split up the `||` into a chain of `switch`es...

```
if (ally.ability === 'Fairy Aura') {
  if (move.type === 'Fairy') auraBoosted = 'Fairy Aura';
} else if (ally.ability === 'Dark Aura') {
  if (move.type === 'Dark') auraBoosted = 'Dark Aura';
} else if (ally.ability === 'Aura Break') {
  auraBroken = true;
} else if (ally.ability === 'Battery') {
  if (ally !== pokemon && move.category === 'Special') {
    // ...
```